### DISCUSSION TITLE
Why does `PositiveIndexKernel` normalize by the target task's variance?

### DISCUSSION THREAD

**Discussion by kalama-ai** [meta-pytorch/botorch#3309] [Q&A]
Dear botorch team,                                                                                                                     

We're using `PositiveIndexKernel` in our transfer learning pipeline and have a few questions about the design choices, specifically around the normalization step.

We understand and appreciate the motivation for positive correlations as described in https://github.com/meta-pytorch/botorch/discussions/3065.      

However, we have questions about the additional normalization by the target task's diagonal entry in                               https://github.com/meta-pytorch/botorch/blob/main/botorch/models/kernels/positive_index.py:
```                                                                                                                                       
if self.unit_scale_for_target:                                                                                                       
      norm_factor = covar[..., self.target_task_index, self.target_task_index]                                                         
      covar = covar / norm_factor.unsqueeze(-1).unsqueeze(-1)  
```                                                                                                                                                            
                                                                                                                                       
  Our questions:
                                                                                                                                       
  1. We notice that `gpytorch`'s `IndexKernel` doesn't include a similar normalization. Is there a specific reason the normalization was introduced together with the positivity constraint rather than being a general improvement applicable to any task covariance kernel?                    
  2. Are the positivity constraint and the normalization truly orthogonal? We'd like to confirm that the positive correlation mechanism works correctly without the normalization step (i.e., with `unit_scale_for_target=False`). Are there any numerical stability or optimization concerns when skipping the normalization?
  3. Is there empirical evidence for the normalization helping? The study in the discussion above demonstrates the benefit of positive correlations, but doesn't isolate the effect of the normalization. Is there evidence that the normalization improves optimization convergence or model quality beyond what the positivity constraint alone provides?
                                                                                                                                       
Context: We're considering using the positivity constraint without the normalization in our framework, and want to make sure we're not losing an important property.

Thank you!
Kathrin

**Comment by hvarfner** [comment]
Hi @kalama-ai ,

Thanks for reaching out!
1. The unit scale was added in order to be in line with the removal of the ScaleKernel, which is the default for our other models. There is no specific reason why the IndexKernel doesn't have it. GPyTorch changes are simply a bit more work on our end. My observations when testing the `unit_scale_for_target` were the same as for the SingleTaskGP: The inclusion of an outputscale may boost optimization performance on specific problems, but (a) doesn't improve things on average, and (b) almost always leads to a much more exploitative, often self-reinforcing, behavior. I don't have ready-to-share results on that specific ablation, though.
2. To be honest, I'm not sure there are ever truly orthogonal changes in BO. The model affects candidate selection, which affects hyperparameter optimization etc. If you come over any interaction or side-effects worth sharing, I would love to see them! The changes were not made for the purpose of numerical stability, but to guard against outputscale shrinkage.
3. I would have to refer you to https://arxiv.org/pdf/2402.02229 for the non-TL case here regarding outputscale shrinkage and overexploitation.

So in short: A free-floating outputscale for the target task is fine, but I personally wouldn't recommend it.

**Comment by kalama-ai** [comment]
Hi @hvarfner,
                                                                                                                                                                           
Thanks a lot for the explanation. We weren’t fully aware of the removal of `ScaleKernel` across your models.                             
                  
One follow-up questions, if you don't mind: We understand that `unit_scale_for_target=True` is the default with a single designated target task. In our framework, users can sometimes have multiple target tasks in a transfer learning setting. Is there a mechanism or recommendation for how to handle normalization when multiple target tasks are involved? Would you simply disable it in that case?

Thanks again for your time!
